### PR TITLE
Sl/pandas shape

### PR DESCRIPTION
**Pull request by sglyon**
@cc7768  will you please review these changes this morning?

### CODE REVIEW

**Comment by sglyon** [comment]
I will leave comments in the file to show you things that I actually changed (instead of what was changed by the pre-commit hook). You only need to look at the sections I comment on

File: Code/notebooks/bootcamp_pandas-shape.ipynb [hunk 1]
```diff
@@ -1,1296 +1,1356 @@
...
+    "\n", 
+    "**Note: requires internet access to run.**  \n", 
+    "\n", 
+    "<!-- \n", 
+    "internal links http://sebastianraschka.com/Articles/2014_ipython_internal_links.html\n", 
+    "-->\n", 
+    "\n", 
+    "This Jupyter notebook was created by Dave Backus, Chase Coleman, and Spencer Lyon for the NYU Stern course [Data Bootcamp](http://databootcamp.nyuecon.com/).  \n", 
+    "\n", 
+    "## tl;dr\n",
```
Comment (sglyon) [inline Code/notebooks/bootcamp_pandas-shape.ipynb]: I added this tl;dr section as a brief summary of the main tools from the chapter.

I'm not 100% sure I like it, what do you think?

File: Code/notebooks/bootcamp_pandas-shape.ipynb [hunk 2]
```diff
@@ -1,1296 +1,1356 @@
...
     "Knowing the order of the index components and being able to inspect their values and names is fundamental to working with a multi-index."
-   ]
-  },
+   ], 
+   "cell_type": "markdown", 
+   "metadata": {}
+  }, 
+  {
+   "source": [
+    "**Exercise**: What would happen if we had switched the order of the strings in the list when we called `dd.set_index`? Try it with this list to find out: `['ISO', 'Country', 'Variable', 'Description', 'Units']`"
```
Comment (sglyon) [inline Code/notebooks/bootcamp_pandas-shape.ipynb:428]: I added this exercise

File: Code/notebooks/bootcamp_pandas-shape.ipynb [hunk 3]
```diff
@@ -1,1296 +1,1356 @@
...
-   },
+    "collapsed": false
+   }
+  }, 
+  {
    "source": [
-    "**Exercise.**  For the dataframe `ddi`:  \n",
-    "\n",
-    "* Use the `reset_index` method to move the `Units` level of the index to a column of the dataframe.\n",
+    "**Exercise.**  For the dataframe `ddi` do the following in separate code cells:  \n",
```
Comment (sglyon) [inline Code/notebooks/bootcamp_pandas-shape.ipynb]: I made sure the exercise was clear to do the list of operations in separate cells

File: Code/notebooks/bootcamp_pandas-shape.ipynb [hunk 4]
```diff
@@ -1,1296 +1,1356 @@
...
+   }
+  }, 
+  {
    "source": [
-    "**Exercise.** With the dataframe `ddt`: \n",
-    "\n",
-    "* What type of object is `Debt`? \n",
+    "**Exercise.** With the dataframe `ddt`: \n", 
+    "\n", 
+    "* What type of object is `ddt[\"Debt\"]`? \n",
```
Comment (sglyon) [inline Code/notebooks/bootcamp_pandas-shape.ipynb]: I made this a bit more clear that we want them to consider `ddt["Debt"]`

File: Code/notebooks/bootcamp_pandas-shape.ipynb [hunk 5]
```diff
@@ -1,1296 +1,1356 @@
...
+   "cell_type": "markdown", 
+   "metadata": {}
+  }, 
   {
-   "cell_type": "code",
-   "execution_count": null,
+   "execution_count": null, 
+   "cell_type": "code", 
+   "source": [
+    "ddts[\"Argentina\"].plot(subplots=True)"
```
Comment (sglyon) [inline Code/notebooks/bootcamp_pandas-shape.ipynb]: I need to delete this as it is the answer to the exercise...

File: Code/notebooks/bootcamp_pandas-shape.ipynb [hunk 6]
```diff
@@ -1,1296 +1,1356 @@
...
+   "cell_type": "markdown", 
+   "metadata": {}
+  }, 
+  {
+   "source": [
+    "## Stacking and unstacking \n", 
+    "\n", 
+    "The `set_index` and `reset_index` methods work on the row labels -- the index.  They move columns to the index and the reverse.  The `stack` and `unstack` methods move index levels to and from column levels:  \n", 
+    "\n", 
+    "* `stack` moves the \"inner most\" (closest to the data when printed) column label into a row label. This creates a **long** dataframe. \n",
```
Comment (sglyon) [inline Code/notebooks/bootcamp_pandas-shape.ipynb]: I re-worded the description of `stack` and and `unstack`

File: Code/notebooks/bootcamp_pandas-shape.ipynb [hunk 7]
```diff
@@ -1,1296 +1,1356 @@
...
+    "ddi.unstack()"
+   ], 
+   "outputs": [], 
+   "metadata": {
+    "collapsed": false
+   }
+  }, 
+  {
+   "source": [
+    "Let's get a smaller subset of this data to work with so we can see things a bit more clearly"
```
Comment (sglyon) [inline Code/notebooks/bootcamp_pandas-shape.ipynb:894]: I added this comment

**Comment by sglyon** [comment]
So the diff isn't actually too bad to look at. 

Things to watch out for:
- I added a `tl;dr` summary section to the top of the shape notebook
- I re-worded our initial description of the stack and unstack methods to be consistent with how I think about them. 

I personally like both changes, but am not sure everyone will.

File: Code/notebooks/bootcamp_pandas-shape.ipynb (hunk 1 above)
Comment (cc7768) [reply Code/notebooks/bootcamp_pandas-shape.ipynb]: I think I like it -- It presents the key ideas. It gives the students an idea of what things they should take away from this notebook

File: Code/notebooks/bootcamp_pandas-shape.ipynb (hunk 2 above)
Comment (cc7768) [reply Code/notebooks/bootcamp_pandas-shape.ipynb:428]: Good exercise. Order matters.

**Comment by cc7768** [comment]
I'm in favor of both changes. Reviewed notebook and I think things look good.

We can merge in a couple hours if no one else has comments.

**Pull request merged**